views/pages/ConfigurationsPage/index.jsx

```diff
@@ -0,0 +1,157 @@
+      }
+      await api.userConfigurations.update(
+        uid,
+        configuration._id,
+        configAttributes
+      )
+
+      await loadAllConfigurations(uid)
+
+      closeForm()
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const loadAllConfigurations = async (userId) => {
+    try {
+      const configurations = await api.userConfigurations.all(userId)
+
+      setConfigurations(configurations)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const closeForm = () => setConfiguration(null)
+
+  const handleFormModal = async (config) => setConfiguration(config)
+
+  const handleDefaultConfiguration = async (configurationId) => {
+    try {
+      const userAttributes = {
+        ...user,
+        preferences: {
+          ...preferences,
+          config: preferences.config === configurationId ? '' : configurationId,
+        },
+      }
+      const updatedUser = await api.users.update(uid, userAttributes)
+
+      setUser(updatedUser)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const handleEditConfig = (configurationId) =>
+    navigate(routes.editConfiguration(configurationId))
+  const handleDeleteConfig = async (configurationId) => {
+    try {
+      await api.userConfigurations.destroy(uid, configurationId)
+
+      await loadAllConfigurations(uid)
+
+      if (preferences.config === configurationId)
+        handleDefaultConfiguration(configurationId)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+  const handleDuplicateConfiguration = async (configuration) => {
+    try {
+      const { _id, ...configAttributes } = configuration
+      const newConfig = {
+        ...configAttributes,
+        owner: uid,
+        readers: [uid],
+      }
+
+      await api.userConfigurations.create(uid, newConfig)
+      await loadAllConfigurations(uid)
+    } catch (error) {
+      handleNotification(error.message)
+    }
+  }
+
+  return (
+    <Box sx={{ p: '20px' }}>
+      <PageHeader
+        title="Configurations"
+        cta={
+          <Button
+            component={Link}
+            endIcon={<Add />}
+            to={routes.newConfiguration}
+            size="small"
+            sx={{ backgroundColor: 'primary.dark', textTransform: 'none' }}
+            variant="contained"
+          >
+            New Configuration
+          </Button>
+        }
+      />
+      <ConfigurationsTable
+        configurations={configurations}
+        onShare={handleFormModal}
+        onDefaultChange={handleDefaultConfiguration}
+        onEdit={handleEditConfig}
+        onDelete={handleDeleteConfig}
+        onDuplicate={handleDuplicateConfiguration}
+        user={user}
+      />
+      {configuration?._id && formValues.length && (
+        <ShareConfigurationForm
+          formValues={formValues}
+          onClose={closeForm}
+          onSubmit={onSubmit}
+          open={open}
```

Comment:
do we need to pass this? I think it will always be `true`.

Reply:
it shouldn't be true always... and we do need it, the form is wrapped by a Dialog box

Reply:
it will be true whenever the condition for rendering this form is true.